solutions/project/project-sandbox/network/nat.yaml

```diff
@@ -0,0 +1,79 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Cloud NAT northamerica-northeast1
+apiVersion: compute.cnrm.cloud.google.com/v1beta1
+kind: ComputeRouterNAT
+metadata:
+  name: project-id-nane1-nat # kpt-set: ${project-id}-nane1-nat
+  namespace: networking # kpt-set: ${namespace}
+  annotations:
+    cnrm.cloud.google.com/project-id: project-id # kpt-set: ${project-id}
+    config.kubernetes.io/depends-on: compute.cnrm.cloud.google.com/namespaces/networking/ComputeRouter/project-id-nane1-router # kpt-set: compute.cnrm.cloud.google.com/namespaces/networking/ComputeRouter/${project-id}-nane1-router
+spec:
+  resourceID: nane1-nat # kpt-set: nane1-nat
```

Comment:
is this `# kpt-set` needed since it doesn't contain any setter?

same thing throughout this file and in subnet.yaml

Reply:
${project-id} is a setter. let me know if I'm misinterpreting your question

Reply:
Could it be that you're missing the ${project-id} in `kpt-set:`? -> `# kpt-set: ${project-id}-nane1-nat`

Reply:
I am pretty sure that the metadata name must match the resourceID. This also applies to other references in this package.

resourceID
Optional
string
...
Immutable. Optional. The name of the resource. Used for creation and acquisition. When unset, the value of `metadata.name` is used as the default.

Reply:
you are right luc, I thought it was a comment for the other line. When resource id is not specified, the resource name in gcp will have the same value as the metadata name. I'm setting a very complex metadata name to ensure uniqueness accross the k8s namespace but the resource name only need to be unique within it's project so I'm defining a much simpler resource name. As you pointed out, I will remove the kpt-set because it is not required.